planner/optimizer.go

```diff
@@ -628,6 +744,52 @@ func operandCanUseIndex(indexType document.ValueType, path document.Path, fc dat
 	return converted, indexType == converted.Type, nil
 }
 
+func getRangesFromFilterNodes(fnodes []*filterNode, indexArity int) (stream.Ranges, error) {
+	if indexArity <= 1 {
+		op := fnodes[0].f.E.(expr.Operator)
+		return getRangesFromOp(op, fnodes[0].v)
+	}
+
+	vb := document.NewValueBuffer()
+	for _, fno := range fnodes {
+		op := fno.f.E.(expr.Operator)
+		v := fno.v
+
+		switch op.(type) {
+		case *expr.EqOperator, *expr.GtOperator, *expr.GteOperator, *expr.LtOperator, *expr.LteOperator:
+			vb = vb.Append(v)
+		case *expr.InOperator:
+			// an index like idx_foo_a_b on (a,b) and a query like
+			// WHERE a IN [1, 1] and b IN [2, 2]
+			// would lead to [1, 1] x [2, 2] = [[1,1], [1,2], [2,1], [2,2]]
+			// which could eventually be added later.
+			panic("unsupported operator IN for composite indexes")
```

Comment:
Once merged, we should track this with an issue, at least not to forget it when making a release 😄